From 17940228e5019ae54bee1865ff01783bb47c5327 Mon Sep 17 00:00:00 2001
From: Dennis M. Pöpperl <free-software@dm-poepperl.de>
Date: Mon, 22 Jul 2019 12:39:02 +0200
Subject: [PATCH] Remove manpage support

---
 misc/Makefile | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/misc/Makefile b/misc/Makefile
index 985541fdd..3e60717a7 100644
--- a/misc/Makefile
+++ b/misc/Makefile
@@ -9,19 +9,13 @@ ifdef DESTDIR
 SETUPTOOLSOPTS = --root="$(DESTDIR)"
 endif
 
-all: man
+all:
 	$(PYTHON) setup.py build
 
-man: doc/qutebrowser.1
-doc/qutebrowser.1: doc/qutebrowser.1.asciidoc
-	a2x -f manpage $<
-
-install: man
+install:
 	$(PYTHON) setup.py install --prefix="$(PREFIX)" --optimize=1 $(SETUPTOOLSOPTS)
 	install -Dm644 misc/org.qutebrowser.qutebrowser.appdata.xml \
 		"$(DESTDIR)$(DATADIR)/metainfo/org.qutebrowser.qutebrowser.appdata.xml"
-	install -Dm644 doc/qutebrowser.1 \
-		"$(DESTDIR)$(MANDIR)/man1/qutebrowser.1"
 	install -Dm644 misc/org.qutebrowser.qutebrowser.desktop \
 		"$(DESTDIR)$(DATADIR)/applications/org.qutebrowser.qutebrowser.desktop"
 	$(foreach i,$(ICONSIZES),install -Dm644 "icons/qutebrowser-$(i)x$(i).png" \
--
libgit2 0.27.8
@@ -35,4 +29,4 @@ install: man
 		  scripts/testbrowser scripts/asciidoc2html.py scripts/setupcommon.py \
 			scripts/link_pyqt.py,$(wildcard scripts/*))
 
-.PHONY: all man install
+.PHONY: all install

